tests_e2e/orchestrator/lib/agent_test_suite_combinator.py

```diff
@@ -0,0 +1,119 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+import logging
+
+from dataclasses import dataclass, field
+from typing import Any, Dict, List, Optional, Type
+
+# E0401: Unable to import 'dataclasses_json' (import-error)
+from dataclasses_json import dataclass_json  # pylint: disable=E0401
+
+# Disable those warnings, since 'lisa' is an external, non-standard, dependency
+#     E0401: Unable to import 'lisa' (import-error)
+#     etc
+from lisa import schema  # pylint: disable=E0401
+from lisa.combinator import Combinator  # pylint: disable=E0401
+from lisa.util import field_metadata  # pylint: disable=E0401
+
+from tests_e2e.orchestrator.lib.agent_test_loader import AgentTestLoader
+
+
+@dataclass_json()
+@dataclass
+class AgentTestSuitesCombinatorSchema(schema.Combinator):
+    test_suites: str = field(
+        default_factory=str, metadata=field_metadata(required=True)
+    )
+
+
+class AgentTestSuitesCombinator(Combinator):
```

Comment:
when is this invoked?

Reply:
It is invoked by LISA.

LISA will invoke the combinator, whose output feeds the the values needed to create the test VM

```
platform:
  - type: azure
     ....
      azure:
        marketplace: $(marketplace_image)
        vhd: $(vhd)
        location: $(location)
        vm_size: $(vm_size)
```